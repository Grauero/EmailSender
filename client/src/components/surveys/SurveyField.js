import React from 'react';

const SurveyField = ({ input, label, meta: { error, touched } }) => (
  <div>
    <label>{label}</label>
    <input {...input} style={{ marginBottom: '5px' }} />
    {touched && error}
  </div>
);

export default SurveyField;
