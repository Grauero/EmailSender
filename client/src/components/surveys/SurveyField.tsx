import React from 'react';

export interface ISurveyField {
  input: object;
  label: string;
  meta: {
    error: string;
    touched: boolean;
  };
}

const SurveyField: React.FC<ISurveyField> = props => {
  const {
    input,
    label,
    meta: { error, touched }
  } = props;

  return (
    <div>
      <label>{label}</label>
      <input {...input} className="input" />
      <div className="red-text touched">{touched && error}</div>
    </div>
  );
};

export default SurveyField;
