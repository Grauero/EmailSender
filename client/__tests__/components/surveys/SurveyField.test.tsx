import 'jest';
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import SurveyField from '../../../src/components/surveys/SurveyField';

const props = {
  input: {},
  label: 'label',
  meta: { error: 'error', touched: true }
};
const component = shallow(<SurveyField {...props} />);

test('matches snapshot', () => expect(toJSON(component)).toMatchSnapshot());
