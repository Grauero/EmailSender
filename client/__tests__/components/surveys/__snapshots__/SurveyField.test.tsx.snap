// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <label>
    label
  </label>
  <input
    className="input"
  />
  <div
    className="red-text touched"
  >
    error
  </div>
</div>
`;
